### PR TITLE
Moved was using the wrong equatable

### PR DESCRIPTION
**Pull request by raulriera**
Small fix that has been lurking around for a while. Thanks to Toussaint Stewart for catching this!

### CODE REVIEW

File: Sources/FunctionalTableData/TableSectionChangeSet.swift [hunk 1]
```diff
@@ -11,10 +11,6 @@ import Foundation
 struct Moved<T: Equatable>: Equatable {
 	let from: T
 	let to: T
-
-	static func ==(lhs: Moved<T>, rhs: Moved<T>) -> Bool {
-		return lhs.from == lhs.from && lhs.to == rhs.to
```
Comment (raulriera) [inline Sources/FunctionalTableData/TableSectionChangeSet.swift]: That `lhs.from == lhs.from`

Comment (raulriera) [reply Sources/FunctionalTableData/TableSectionChangeSet.swift]: Unsure about a test for this 🤔 how come no test changed

**Comment by raulriera** [comment]
Stop it bot

File: Sources/FunctionalTableData/TableSectionChangeSet.swift (hunk 1 above)
Comment (dannbeau) [reply Sources/FunctionalTableData/TableSectionChangeSet.swift]: I am wondering if it has anything to do with https://github.com/Shopify/store/issues/11412
We have been tracking down this issue that ultimately creates the inconsistent table data exception but the change set is correct. I have to deep dive on how this struct is used.

**Pull request merged**